ctk/common/src/main/kotlin/org/codice/compliance/verification/core/assertions/StatementVerifier.kt

```diff
@@ -0,0 +1,156 @@
+                        .any { it.children(SUBJECT).isEmpty() })
+            throw SAMLComplianceException.create(SAMLCore_2_7_2,
+                    message = "An AuthnStatement was found without a Subject element.",
+                    node = node)
+    }
+
+    /** 2.7.2.1 Element <SubjectLocality> **/
+    private fun verifySubjectLocality() {
+        node.allChildren("SubjectLocality").forEach {
+            val address = it.attributes.getNamedItem("Address")
+            val dNSName = it.attributes.getNamedItem("DNSName")
+
+            if (address != null) CommonDataTypeVerifier.verifyStringValues(address)
+            if (dNSName != null) CommonDataTypeVerifier.verifyStringValues(dNSName)
+        }
+    }
+
+    /** 2.7.2.2 Element <AuthnContext> **/
+    private fun verifyAuthnContext() {
+        node.allChildren("AuthnContext").forEach {
+            val authnContextClassRef = it.attributes.getNamedItem("AuthnContextClassRef")
+            val authnContextDeclRef = it.attributes.getNamedItem("AuthnContextDeclRef")
+            val authnContextDecl = it.attributes.getNamedItem("AuthnContextDecl")
+            val authenticatingAuthority = it.attributes.getNamedItem("AuthenticatingAuthority")
+
+            if (authnContextClassRef != null)
+                CommonDataTypeVerifier.verifyUriValues(authnContextClassRef)
+            if (authnContextDeclRef != null)
+                CommonDataTypeVerifier.verifyUriValues(authnContextDeclRef)
+            if (authnContextDecl != null)
+                CommonDataTypeVerifier.verifyUriValues(authnContextDecl)
+            if (authenticatingAuthority != null)
+                CommonDataTypeVerifier.verifyUriValues(authenticatingAuthority)
+        }
+    }
+
+    /** 2.7.3 Element <AttributeStatement> **/
+    private fun verifyAttributeStatement() {
+        if (node.allChildren(ASSERTION)
+                        .filter { it.children("AttributeStatement").isNotEmpty() }
+                        .any { it.children(SUBJECT).isEmpty() })
+            throw SAMLComplianceException.create(SAMLCore_2_7_3,
+                    message = "An AttributeStatement was found without a Subject element.",
+                    node = node)
+    }
+
+    /** 2.7.3.1 Element <Attribute> **/
+    private fun verifyAttribute() {
+        node.allChildren("Attribute").forEach {
+            val name = it.attributes.getNamedItem("Name")
+            val nameFormat = it.attributes.getNamedItem("NameFormat")
+            val friendlyName = it.attributes.getNamedItem("FriendlyName")
+
+            if (name != null) CommonDataTypeVerifier.verifyStringValues(name)
+            if (nameFormat != null) CommonDataTypeVerifier.verifyUriValues(nameFormat)
+            if (friendlyName != null) CommonDataTypeVerifier.verifyStringValues(friendlyName)
+        }
+    }
+
+    /** 2.7.3.2 Element <EncryptedAttribute> **/
+    private fun verifyEncryptedAttribute() {
```

Comment:
❗️ This verification can be removed. Covered in EncryptionVerifier.kt in an upcoming PR: https://github.com/connexta/saml-conformance/pull/92/files#diff-ea01368ee04c99da2848408cd4d34ea3R65